lib/rust/parser/src/data/tree.rs

```diff
@@ -0,0 +1,214 @@
+//! A map-based tree for use in the macro resolver.
```

Comment:
Can you move this struct out of parser to our common libs please? Its super useful